client/trino-cli/src/main/java/io/trino/cli/Console.java

```diff
@@ -102,7 +106,16 @@ public Integer call()
 
     public boolean run()
     {
-        ClientSession session = clientOptions.toClientSession();
+        // map option names to URL params and restrict those set explicitly by the user
```

Comment:
extract this logic to a separate method `private static List<String> getCommandLineArguments(CommandSpec spec)`